conda/gateways/repodata/__init__.py

```diff
@@ -579,7 +579,9 @@ def load_state(self):
         """
         try:
             self.load(state_only=True)
-        except FileNotFoundError:  # or JSONDecodeError?
+        except (FileNotFoundError, json.JSONDecodeError) as e:
+            if isinstance(e, json.JSONDecodeError):
+                warnings.warn(f"{e.__class__.__name__} loading {self.cache_path_state}")
```

Comment:
Is a warning the right thing here? Should we use a debug logger?

```suggestion
        except (FileNotFoundError, json.JSONDecodeError) as exc:
            if isinstance(exc, json.JSONDecodeError):
                warnings.warn(f"{exc.__class__.__name__} loading {self.cache_path_state}")
```

Reply:
> `e` is the idiomatic variable name for an exception.

Agree to disagree, descriptive variable names are also idiomatic, make the code more accessible for beginners.

> A logger could be appropriate.

Should the logger be a separate PR or should there be a follow-up ticket?

Reply:
The two main choices for this symbol that is usually defined on one line and only used on the following line appear to be `e` and a random 2-5 character string starting with the letter `e`. In conda 136 are `e` out of 199 total `except .* as`.

https://github.com/search?type=code&q=%2Fexcept+.*+as+e%3A%2F+language%3APython

https://github.com/search?q=org%3Aconda+%2Fexcept+.*+as+%5Cw%2B%3A%2F&type=code

Reply:
Thanks for the additional data!